### PR TITLE
Implement Rust Wasm Plugin Build & Publish Action

### LINKED ISSUE #1469: Implement Rust Wasm Plugin Build & Publish Action
## Why do you need it?

We have supported the Build & Publish Action for Go Wasm plugins:

https://github.com/alibaba/higress/blob/main/.github/workflows/build-and-push-wasm-plugin-image.yaml

Need to extend this to support the Build & Publish of Rust Wasm plugins

### PR DESCRIPTION
**Pull request by 007gzs**
### Ⅱ. Does this pull request fix one issue?

fixes #1469

### CODE REVIEW

File: .github/workflows/build-and-push-wasm-plugin-image.yaml [hunk 1]
```diff
@@ -23,32 +30,40 @@ jobs:
     env:
       IMAGE_REGISTRY_SERVICE: ${{ vars.IMAGE_REGISTRY || 'higress-registry.cn-hangzhou.cr.aliyuncs.com' }}
       IMAGE_REPOSITORY: ${{ vars.PLUGIN_IMAGE_REPOSITORY || 'plugins' }}
+      RUST_VERSION: 1.82
       GO_VERSION: 1.19
       TINYGO_VERSION: 0.28.1
       ORAS_VERSION: 1.0.0
     steps:
-      - name: Set plugin_name and version from inputs or ref_name
+      - name: Set plugin_type、 plugin_name and version from inputs or ref_name
```
Comment (jizhuozhi) [inline .github/workflows/build-and-push-wasm-plugin-image.yaml:38]: don't use full width character in english comments

File: plugins/wasm-rust/DockerfileBuilder [hunk 2]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (jizhuozhi) [inline plugins/wasm-rust/DockerfileBuilder:1]: why do we need another Dockerfile to build?

Comment (007gzs) [reply plugins/wasm-rust/DockerfileBuilder:1]: I referred to the builder used for packaging and publishing in the go-lang https://github.com/alibaba/higress/blob/main/plugins/wasm-go/DockerfileBuilder

File: .github/workflows/build-and-push-wasm-plugin-image.yaml (hunk 1 above)
Comment (007gzs) [reply .github/workflows/build-and-push-wasm-plugin-image.yaml]: done

File: plugins/wasm-rust/DockerfileBuilder (hunk 2 above)
Comment (jizhuozhi) [reply plugins/wasm-rust/DockerfileBuilder:1]: Dockerfile is enough, no need additional files

Comment (jizhuozhi) [reply plugins/wasm-rust/DockerfileBuilder:1]: Both Dockerfile and DockerfileBuilder are for building an artifact named plugin.wasm, but the oras is a client which aims to publish artifact to registry. It means no matter which language is, the only different is how to build an artifact but using same way to publish artifact. 

In a word, I think the oras should be a independent image, but not mixed in builders. 

cc @johnlanni

Comment (johnlanni) [reply plugins/wasm-rust/DockerfileBuilder:1]: builder mainly facilitates developers by eliminating the need to download corresponding versions of oras, as well as Rust build tools, etc

Comment (jizhuozhi) [reply plugins/wasm-rust/DockerfileBuilder:1]: For developers, devcontainer might be the best solution and I still think it should keep simple for OCI building

Comment (johnlanni) [reply plugins/wasm-rust/DockerfileBuilder:1]: Well, you're right, developers just need the Dockerfile, but it seems that the DockerfileBuilder used for building images allows the workflow for publishing wasm plugins to remain consistent with Go. From this perspective, it still makes sense

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**